Wikipedia/Code/TalkPageFetcher.swift

```diff
@@ -1,227 +1,68 @@
-import WMF
+struct TalkPageItem: Codable {
+    let type: TalkPageItemType
+    let level: Int?
+    let id: String?
+    let html: String?
+    let headingLevel: Int?
+    let placeholderHeading: Bool?
+    let replies: [TalkPageItem]
+    let otherContent: String?
 
-enum TalkPageType: Int {
-    case user
-    case article
     
-    func canonicalNamespacePrefix(for siteURL: URL) -> String? {
-        let namespace: PageNamespace
-        switch self {
-        case .article:
-            namespace = PageNamespace.talk
-        case .user:
-            namespace = PageNamespace.userTalk
-        }
-        return namespace.canonicalName + ":"
-    }
-    
-    func titleWithCanonicalNamespacePrefix(title: String, siteURL: URL) -> String {
-        return (canonicalNamespacePrefix(for: siteURL) ?? "") + title
-    }
-    
-    func titleWithoutNamespacePrefix(title: String) -> String {
-        if let firstColon = title.range(of: ":") {
-            var returnTitle = title
-            returnTitle.removeSubrange(title.startIndex..<firstColon.upperBound)
-            return returnTitle
-        } else {
-            return title
-        }
+    enum CodingKeys: String, CodingKey {
+        case type, level, id, html,headingLevel, placeholderHeading, replies
+        case otherContent = "othercontent"
     }
     
-    func urlTitle(for title: String) -> String? {
-        assert(title.contains(":"), "Title must already be prefixed with namespace.")
-        return title.denormalizedPageTitle
+    enum TalkPageItemType: String, Codable {
+        case comment = "comment"
+        case heading = "heading"
     }
 }
 
-enum TalkPageFetcherError: Error {
-    case talkPageDoesNotExist
-}
-
 class TalkPageFetcher: Fetcher {
     
-    private let sectionUploader = WikiTextSectionUploader()
-    
-    func addTopic(to title: String, siteURL: URL, subject: String, body: String, completion: @escaping (Result<[AnyHashable : Any], Error>) -> Void) {
-        
-        guard let url = postURL(for: title, siteURL: siteURL) else {
-            completion(.failure(RequestError.invalidParameters))
-            return
-        }
-        
-        sectionUploader.addSection(withSummary: subject, text: body, forArticleURL: url) { (result, error) in
-            if let error = error {
-                completion(.failure(error))
-                return
-            }
-            
-            guard let result = result else {
-                completion(.failure(RequestError.unexpectedResponse))
-                return
-            }
-            
-            completion(.success(result))
-        }
-    }
-    
-    func addReply(to topic: TalkPageTopic, title: String, siteURL: URL, body: String, completion: @escaping (Result<[AnyHashable : Any], Error>) -> Void) {
-        
-        guard let url = postURL(for: title, siteURL: siteURL) else {
+    func fetchTalkPageContent(url: URL, completion: @escaping (Result<[TalkPageItem], Error>) -> Void) {
```

Comment:
Since we're moving towards the empty view controller taking in a `siteURL` and `talkPageTitle` in the feature flag PR, change this url into two params : `talkPageTitle: String, siteURL: URL`. Then you can remove this guard and use them when building the request below:

```
...
        let params = ["action" : "discussiontoolspageinfo",
                      "page" : talkPageTitle.denormalizedPageTitle,
                      "format": "json",
                      "prop" : "threaditemshtml",
                      "fomatversion" : "2"
        ]

        performDecodableMediaWikiAPIGET(for: siteURL, with: params) { (result: Result<TalkPageAPIResponse, Error>) in
...
```

`denormalizedPageTitle` is just our helper extension to replace spaces with an underscore - that's how the API wants titles.